specs/Checkbox.md

```diff
@@ -0,0 +1,388 @@
+# Checkbox component specification
+
+The `Checkbox` component allows a user to choose between two mutually exclusive options. 
+
+## Related variant considerations
+
+- Toggle: should be a separate component because it has different anatomy & warrants unique themability.
+
+- Indeterminate/Tri State: should support groups with hierarchical checkboxes where the parent checkbox can be in a mixed state if its children checkboxes aren't all checked (and if they are all checked, then the parent is checked; same if they are all unchecked, then the parent is unchecked).
+
+## Reference implementations
+
+https://codesandbox.io/s/checkboxes-ggpx1
```

Comment:
Would be awesome to have a page like this in the repo where devs can do in depth research on how other libraries think about the same component.